test/sing_for_needs_web/schema/query/causes_test.exs

```diff
@@ -181,4 +181,92 @@ defmodule SingForNeeds.Schema.Query.CauseTest do
 
     assert expected_result == json_response(conn, 200)
   end
+
+  test "causes ending soon and limit passed to causes query" do
+    causes_ending_soon_with_limit_query = """
+      query($scope: String, $limit: Int) {
+          causes(scope: $scope, limit: $limit) {
+              name
+          }
+      }
+    """
+
+    twenty_days_from_now = Timex.add(Timex.now(), Timex.Duration.from_days(20))
+    thirty_days_from_now = Timex.add(Timex.now(), Timex.Duration.from_days(30))
+    fifteen_days_from_now = Timex.add(Timex.now(), Timex.Duration.from_days(15))
+    five_days_ago = Timex.add(Timex.now(), Timex.Duration.from_days(-5))
+    nine_days_ago = Timex.add(Timex.now(), Timex.Duration.from_days(-9))
+
+    insert(:cause, %{name: "Cause ends in 30 days", end_date: thirty_days_from_now})
+    insert(:cause, %{name: "Cause ends in 20 days", end_date: twenty_days_from_now})
+    insert(:cause, %{name: "Cause ends in 15 days", end_date: fifteen_days_from_now})
+    insert(:cause, %{name: "Cause ended 5 days ago", end_date: five_days_ago})
+    insert(:cause, %{name: "Cause ended 9 days ago", end_date: nine_days_ago})
+
+    conn = build_conn()
+
+    conn =
+      post conn, "/api",
+        query: causes_ending_soon_with_limit_query,
+        variables: %{scope: "ending_soon", limit: 2}
+
+    expected_result = %{
+      "data" => %{
+        "causes" => [
+          %{"name" => "Cause ends in 15 days"},
+          %{"name" => "Cause ends in 20 days"}
+        ]
+      }
+    }
+
+    assert expected_result == json_response(conn, 200)
+  end
+
+  test "scope: trending and limit:2 passed to causes query" do
+    trending_causes_with_limit_query = """
+      query($scope: String, $limit: Int) {
+          causes(scope: $scope, limit: $limit) {
+              name
+          }
+      }
+    """
+
+    artists = insert_list(4, :artist)
+
+    insert(:cause, %{
+      name: "Cause with Medium Donation",
+      amount_raised: 30_000,
+      artists: Enum.take(artists, 2),
+      description: Faker.Lorem.paragraph()
+    })
+
+    insert(:cause, %{name: "Cause with Least Donation", amount_raised: 10_000})
+
+    insert(:cause, %{
+      name: "Cause with Least Donation and two artists",
+      amount_raised: 10_000,
+      artists: Enum.take(artists, -2)
+    })
+
+    insert(:cause, %{name: "Cause with Highest Donation", amount_raised: 90_000, artists: artists})
+
+    conn = build_conn()
+
+    conn =
+      post conn, "/api",
+        query: trending_causes_with_limit_query,
+        variables: %{scope: "trending", limit: 3}
+
+    expected_result = %{
+      "data" => %{
+        "causes" => [
+          %{"name" => "Cause with Highest Donation"},
+          %{"name" => "Cause with Medium Donation"},
+          %{"name" => "Cause with Least Donation and two artists"}
+        ]
+      }
+    }
+
+    assert expected_result == json_response(conn, 200)
```

Comment:
Personally I don't mind the additional overhead of an `actual_result` (or more specifically named) variable as the subject of comparison here.